File: python/tools/dashboards/delete.py
import json

from api.client import is_error, is_success
from schema.tool_inputs import DashboardDeleteSchema
from tools.tool_definitions import get_tool_definition
from tools.types import Context, Tool, ToolResult


async def delete_dashboard_handler(context: Context, params: DashboardDeleteSchema) -> ToolResult:
    project_id = await context.get_project_id()
    result = await context.api.dashboards(project_id).delete(int(params.dashboardId))

    if is_error(result):
        raise Exception(f"Failed to delete dashboard: {result.error}")

    assert is_success(result)
    return ToolResult(content=json.dumps(result.data))


def delete_dashboard_tool() -> Tool[DashboardDeleteSchema]:
    definition = get_tool_definition("dashboard-delete")

    return Tool(
        name="dashboard-delete",
        description=definition["description"],
        schema=DashboardDeleteSchema,
        handler=delete_dashboard_handler,
    )

File: typescript/src/tools/dashboards/delete.ts
import { DashboardDeleteSchema } from "@/schema/tool-inputs";
import { getToolDefinition } from "@/tools/toolDefinitions";
import type { Context, Tool } from "@/tools/types";
import type { z } from "zod";

const schema = DashboardDeleteSchema;

type Params = z.infer<typeof schema>;

export const deleteHandler = async (context: Context, params: Params) => {
	const { dashboardId } = params;
	const projectId = await context.getProjectId();
	const result = await context.api.dashboards({ projectId }).delete({ dashboardId });

	if (!result.success) {
		throw new Error(`Failed to delete dashboard: ${result.error.message}`);
	}

	return { content: [{ type: "text", text: JSON.stringify(result.data) }] };
};

const definition = getToolDefinition("dashboard-delete");

const tool = (): Tool<typeof schema> => ({
	name: "dashboard-delete",
	description: definition.description,
	schema,
	handler: deleteHandler,
});

export default tool;
